caramel-batch/src/main/kotlin/com/whatever/caramel/batch/config/ScheduleNotificationRemoveBatchConfig.kt

```diff
@@ -0,0 +1,100 @@
+package com.whatever.caramel.batch.config
+
+import com.whatever.caramel.domain.notification.model.NotificationType
+import com.whatever.caramel.domain.notification.model.ScheduledNotification
+import com.whatever.caramel.domain.notification.service.ScheduledNotificationService
+import jakarta.persistence.EntityManagerFactory
+import org.springframework.batch.core.BatchStatus
+import org.springframework.batch.core.Job
+import org.springframework.batch.core.JobExecution
+import org.springframework.batch.core.JobExecutionListener
+import org.springframework.batch.core.Step
+import org.springframework.batch.core.configuration.annotation.StepScope
+import org.springframework.batch.core.job.builder.JobBuilder
+import org.springframework.batch.core.launch.support.RunIdIncrementer
+import org.springframework.batch.core.repository.JobRepository
+import org.springframework.batch.core.step.builder.StepBuilder
+import org.springframework.batch.item.ItemReader
+import org.springframework.batch.item.ItemWriter
+import org.springframework.batch.item.database.JpaPagingItemReader
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.transaction.PlatformTransactionManager
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+@Configuration
+class ScheduleNotificationRemoveBatchConfig(
+    private val scheduledNotificationService: ScheduledNotificationService,
+) {
+    @Bean
+    @StepScope
+    fun scheduleRemoveItemReader(entityManagerFactory: EntityManagerFactory): JpaPagingItemReader<ScheduledNotification> {
+        val zoneSource = ZoneId.of("Asia/Seoul")
+        val localDateTime = LocalDateTime.now(zoneSource)
+        val startOfDay = localDateTime.toLocalDate().atStartOfDay(zoneSource).toLocalDateTime()
+        val endOfDay = localDateTime.toLocalDate().atTime(23, 59, 59)
```

Comment:
이렇게도 가능할지도??
endOfDay에 들어가는 withNano(0)은 운 나쁘게 조회 들어갈 경우 overflow 발생할까봐 넣었습니다~
```suggestion
        val localDate = LocalDate.now(zoneSource)
        val startOfDay = localDate.atStartOfDay()
        val endOfDay = localDate.atTime(23, 59, 59).withNano(0)
```

Reply:
@given-dragon 
f0066ec033e4c91a5490fdcbe5fb43e36af2a4b9 로직 수정까지 해주시고,, 감사합니다~ 
withNano(0)이 초단위 까지만 표기해주는 거죠?! 맞나요?
날짜 관련해서는 처음 작업해봐서 잘 몰라서요 ㅎ;; 
예시를 포함해서 좀더 자세하게 설명해주세요!

Reply:
LocalDateTime은 소숫점 9자리까지(나노초 까지) 지원하고, postgresql의 timestamp는 소숫점 6자리까지(마이크로초)까지 지원합니다!
그래서 LocalTime.MAX 값은 `23:59:59.999999999`지만, 데이터베이스에서 반올림을 하며 '다음 날의 00:00:00.000000'로 쿼리가 들어가는 문제가 발생해요!
그래서 나노초를 0으로 세팅해서 반올림이 발생하지 않도록 해줬습니다~

DB 쿼리로도 간단하게 확인할 수 있어요!
<img width="787" height="214" alt="image" src="https://github.com/user-attachments/assets/759ee86e-6a26-44b3-bece-59218cea43ad" />

Reply:
@given-dragon 
굿 ~ 배워갑니다?